packages/devtools_app/lib/src/screens/logging/_log_details.dart

```diff
@@ -89,7 +87,11 @@ class _LogDetailsState extends State<LogDetails>
           needsTopBorder: false,
           rightActions: [
             CopyToClipboardControl(
-              dataProvider: disabled ? null : () => log?.prettyPrinted,
+              dataProvider: disabled
+                  ? null
+                  // The parenthesis are actually needed.
+                  // ignore: unnecessary_parenthesis
+                  : (() => log?.prettyPrinted!) as String Function()?,
```

Comment:
you should be able to get rid of the cast if you write:
`dataProvider: disabled ? null : log?.prettyPrinted`

and then change prettyPrinted from a getter to a function.

Reply:
prettyPrinted can return null, but we want function that  returns non-nullable string for dataProvider. 
Do we want to change dataProvider type to return nullable string?

Still converted prettyPrinted to function as it is too complicated to be getter.

Reply:
Yeah let's change ClipboardDataProvider to return a nullable string. It looks like down the line, ClipboardData.text is nullable anyway, so this should be fine. We might want to ensure we give a default string value for `data.text` in `Clipboard.setData`

Reply:
done